velox/exec/HashProbe.cpp

```diff
@@ -593,8 +657,16 @@ int32_t HashProbe::evalFilter(int32_t numRows) {
       rawMapping[numPassed++] = row;
     };
     for (auto i = 0; i < numRows; ++i) {
-      const bool passed = !decodedFilterResult_.isNullAt(i) &&
-          decodedFilterResult_.valueAt<bool>(i);
+      bool passed;
+      if (nonNullRows_.isValid(i)) {
+        passed = !decodedFilterResult_.isNullAt(i) &&
+            decodedFilterResult_.valueAt<bool>(i);
+        if (!passed) {
+          passed = testFilterOnBuildSide(i, filterRowsBuildSideNulls_);
```

Comment:
We need to review the logic here to leverage default null behavior of the filter